### PR TITLE
Refactor ASTContext::getDeclAlign() (NFC)

### PR DESCRIPTION
**Pull request by JonPsson1**
@efriedma-quic
@rjmccall

While working on this (with the other patch: 72886), I found this refactoring at the top of the function which I like. What do you think?

### CODE REVIEW

File: clang/lib/AST/ASTContext.cpp [hunk 1]
```diff
@@ -1627,28 +1627,20 @@ const llvm::fltSemantics &ASTContext::getFloatTypeSemantics(QualType T) const {
...
+  // __attribute__((aligned)) can increase or decrease alignment
+  // *except* on a struct or struct member, where it only increases
+  // alignment unless 'packed' is also specified.
+  //
+  // It is an error for alignas to decrease alignment, so we can
+  // ignore that possibility;  Sema should diagnose it.
+  bool IsPackedField = isa<FieldDecl>(D) &&
+                       (D->hasAttr<PackedAttr>() ||
+                        cast<FieldDecl>(D)->getParent()->hasAttr<PackedAttr>());
+  bool UseAlignAttrOnly = isa<FieldDecl>(D) ? IsPackedField : AlignFromAttr;
```
Comment (efriedma-quic) [inline clang/lib/AST/ASTContext.cpp:1643]: That's a lot of `isa<>` and `cast<>`. I think I'd prefer something more like:

```
bool UseAlignAttrOnly;
if (FieldDecl *FD = dyn_cast<FieldDecl>(D)) {
  UseAlignAttrOnly = FD->hasAttr<PackedAttr>() ||
                     FD->getParent()->hasAttr<PackedAttr>();
} else {
  UseAlignAttrOnly = AlignFromAttr != 0;
}
```

Comment (JonPsson1) [reply clang/lib/AST/ASTContext.cpp]: That works for me as well: main problem to me with the original code was the duplicated handling of a FieldDecl.

I took your suggestion, but removed the braces, which seem superfluous? Or are they required by the coding standard?

**Comment by JonPsson1** [comment]
Eli, are you supposed to press some button before I merge this?

**Review by efriedma-quic — APPROVED** [review]
LGTM

See https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements for the rules on use/don't use braces.  A single multi-line statement is sort of on the edge; I tend to prefer braces in that case, but you can go either way.

**Pull request merged**